### ISSUE TITLE
Starting application from background with notification happens in foreground (iOS)

### ISSUE THREAD

**Issue by yedidyak** [wix/detox#590]
When trying to test opening an application from the background with a push notification with this code:

```js
await device.sendToHome();
await device.relaunchApp({ newInstance: false, userNotification });
```

The notification is sent to the application but only after the native `UIApplicationState` is already changed to `UIApplicationStateActive`. This makes the application behave as if the notification was opened when the app was already in the foreground.

The same behaviour happens when trying like this:

```js
await device.sendToHome();
await device.sendUserNotification(userNotification);
await device.relaunchApp({newInstance: false});
```

@LeoNatan

**Comment by LeoNatan** [comment]
Thanks

**Comment by LeoNatan** [comment]
Just to prepare you, @yedidyak, you are relying on bad practice in your application. The application states you rely on are Apple implementation detail, and should not be relied upon in production code. For example, you are using an old, deprecated API. Using the new UserNotifications API brings a different implementation and states are different there.

The more I think about the solution to fix this issue (and how much it complicates the system), the more resistant I am to fixing it. Will discuss more with @rotemmiz, but just be prepared for this possible outcome.

**Comment by LeoNatan** [comment]
Further discussed with @rotemmiz. We have decided to mark this as "wontfix" and close. We plan to remove support of this deprecated API in Detox soon anyway.

All is not lost, because we have a guild task to upgrade react-native-notifications to use the new Apple API instead of the old one. So it will be solved one way or another.

**Comment by rotemmiz** [comment]
For the sake of history, this was indeed a bug, which was handled in #604, by changing the way notifications and urls are being sent while application is in background.
